mlrun/model_monitoring/stream_processing.py

```diff
@@ -310,7 +306,10 @@ def apply_infer_schema():
                 table=self.kv_path,
             )
 
-        if self.model_endpoint_store_target == ModelEndpointTarget.V3IO_NOSQL:
+        store_object = mlrun.model_monitoring.get_store_object(
+            project=self.project, secret_provider=secret_provider
+        )
+        if store_object.type == ModelEndpointTarget.V3IO_NOSQL:
             apply_infer_schema()
```

Comment:
Unrelated: do you know why we limit this to V3IO only?

Reply:
yes, because we have to generate V3IO KV Schema for Grafana. In general, these steps should be wrapped within the KV/SQL abstraction (similar to what we did with the TSDB connector) but anyway for now we use that approach.

Reply:
I see, thx